src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -77,6 +77,30 @@ public function __construct(bool $magicCall = false, bool $throwExceptionOnInval
         $this->cacheItemPool = $cacheItemPool instanceof NullAdapter ? null : $cacheItemPool; // Replace the NullAdapter by the null value
     }
 
+    /**
+	 * @param $objectOrArray
+	 * @param $propertyPath
+	 *
+	 * @return null|string
+	 * @throws \ReflectionException
+	 */
+	public function getType(&$objectOrArray, $propertyPath): ? string
+	{
+		$propertyPath = $this->getPropertyPath($propertyPath);
+		$property  = $propertyPath->getElement($propertyPath->getLength() - 1);
+		$access = $this->getWriteAccessInfo(get_class($objectOrArray), $property, []);
```

Comment:
`get_class($objectOrArray)` looks really bad to me. If it is an array rather than an object, it will break.

and this code is totally broken for deep property paths, as you take the last property and apply it on the root object, so this does not work at all.

Reply:
I agree with @stof here, you should be careful if this param is an array and test it. it trigger a warning and return false.

And `getWriteAccessInfo` accept only strings.

Reply:
And I think that 

$property does not need to be a variable and can be inlined in the `getWriteAccessInfo`